bench/app.py

```diff
@@ -277,7 +277,7 @@ def pull_apps(apps=None, bench_path='.', reset=False):
 			if reset:
 				reset_cmd = "git reset --hard {remote}/{branch}".format(remote=remote, branch=branch)
 				if get_config(bench_path).get('shallow_clone'):
-					exec_cmd("git fetch --depth=1 --no-tags {remote} {branch}".format(remote=remote, branch=branch),
+					exec_cmd("git fetch {remote} {branch}".format(remote=remote, branch=branch),
```

Comment:
This change is a bit unclear. Overall, I'm not certain what these changes will change. Can you add what is the expected behaviour of these changes to clear this up?